DeleteS3FeatureBucket/index.js

```diff
@@ -1,75 +1,90 @@
-    const repoName = pullRequest['head']['repo']['name'];
-
-    console.log({action});
-    console.log({branchName});
-
-    if(action === 'closed' && branchName != 'master') {
-        const bucket = `${branchName}-${repoName}`;
-        let objects;
-
-        // List all object keys in the bucket
-        try {
-          let response = await s3.listObjects({Bucket: bucket}).promise();
-          objects = response['Contents'].map((object) => { return { 'Key': object['Key'] } });
-        } catch (e) {
-          console.log(e);
-          throw new Error(e);
-        }
-
-        console.log({objects});
-
-        // Delete all the objects from the bucket
-        await s3.deleteObjects({Bucket: bucket, Delete: { Objects: objects }}).promise();
-        // Delete the bucket
-        await s3.deleteBucket({Bucket: bucket}).promise();
-        message = `S3 bucket deleted: ${bucket}`;
-    } else {
-        message = "Pull request event ignored";
-    }
-    return { statusCode: 200, body: message };
+async function deleteS3Bucket(bucket) {
+  let objects;
+
+  // List all object keys in the bucket
+  try {
+    let response = await s3.listObjects({Bucket: bucket}).promise();
+    objects = response['Contents'].map((object) => { return { 'Key': object['Key'] } });
+  } catch (e) {
+    console.log(e);
+    throw new Error(e);
+  }
+
+  console.log({objects});
+
+  // Delete all the objects from the bucket
+  await s3.deleteObjects({Bucket: bucket, Delete: { Objects: objects }}).promise();
+  // Delete the bucket
+  await s3.deleteBucket({Bucket: bucket}).promise();
+
+  return message = `S3 bucket deleted: ${bucket}`;
+};
+
+function parseWebhookBody(eventType, webhookBody) {
+  let shouldDeleteBucket, branchName, repoName;
+
+  if (eventType === 'pull_request') {
+    branchName = webhookBody['pull_request']['head']['ref'];
+    repoName = webhookBody['pull_request']['head']['repo']['name'];
+    shouldDeleteBucket = webhookBody['action'] === 'closed' && branchName !== 'master';
+  } else if (eventType === 'delete') {
+    branchName = webhookBody['ref'];
+    repoName = webhookBody['repository']['name'];
+    shouldDeleteBucket = webhookBody['ref_type'] === 'branch' && branchName !== 'master';
+  };
+
+  console.log('shouldDeleteBucket: ', shouldDeleteBucket);
+  return { shouldDeleteBucket, branchName, repoName };
 }
 
+async function handleWebhookEvent(eventType, webhookBody) {
+  if (!['pull_request', 'delete'].includes(eventType)) {
+    const message = `Event ignored. Event type: ${eventType}`;
+    return { statusCode: 200, body: message };
+  }
+
+  const { shouldDeleteBucket, branchName, repoName } = parseWebhookBody(eventType, webhookBody);
+
+  if (!shouldDeleteBucket) {
+    const message = `Event should not delete bucket: shouldDeleteBucket: ${shouldDeleteBucket}, eventType: ${eventType}`;
+    return { statusCode: 200, body: message };
+  } else {
+    const bucket = `${branchName}-${repoName}`;
```

Comment:
for hobbes branches, I think we also need to delete the `${branchName}-test` and ${branchName}-dev` buckets because they're not managed by cloudformation

Reply:
ah the hobbes buckets are slightly differently named - I missed that.

Good point about deleting the other env buckets as well. I had completely forgotten we didn't add `-hobbes` and also had the env endings! I will make that change!